### ISSUE TITLE
Desktop Client downloads files initially when virtual files is selected

### ISSUE THREAD

**Issue by valdearg** [nextcloud/desktop#3810]
When syncing using the "Use virtual files instead of downloading content immediately", the desktop client starts to sync data anyway?

### Expected behaviour
When setting up a sync connection in the client, select the option for "Use virtual files instead of downloading content immediately", checking bandwidth usage it's still downloading all the data.

### Actual behaviour
Starts downloading the files instead, when this is 1TB+ it's a bit stupid.

### Steps to reproduce
1. Click on "Add Folder Sync Connection"
2. Choose an empty local folder you want to sync to.
3. Choose a large folder on the server to sync. Leave the checkbox for virtual files.

![nextcloud_2021-09-19_21-50-26](https://user-images.githubusercontent.com/2789033/133943193-44fb7c75-f1e1-44de-abd8-a5e510834108.png)

4. Let it sync. You'll find it starts downloading.

![nextcloud_2021-09-19_21-54-18](https://user-images.githubusercontent.com/2789033/133943221-50157aa4-4979-448d-a0e4-605b7ae0fb02.png)

Or:

![nextcloud_2021-09-19_22-15-48](https://user-images.githubusercontent.com/2789033/133943292-01449c29-0409-4410-aad0-fe8c156f12d4.png)

Monitoring my bandwidth while this is happening, this is downloading at full speed.

### Client configuration
Client version: 3.3.3

Operating system: Windows 10

OS language: English (United Kingdom)

### Logs

1. Client logfile: https://cloud.neko.network/s/2oZoS7rfmCg4r5w

**Comment by valdearg** [comment]
Weirdly enough, if I pause the sync and choose the options > availability > Free up local space option, it'll do the "checking for changes", it'll then create some virtual files and then start downloading again.

**Comment by valdearg** [comment]
Updated to the latest client version 3.3.4, I managed to get one folder fully synced down, however it took around 5 times of doing:

1. Choose the Options > Availability > Free up local space
2. Resume sync. The client then goes through the checking for changes on the server.
3. Speeds through creating a number of files, the first time was around 14800 files. It then arbitrarily decided to start downloading the remaining files.
4. Pause the sync.
5. Close the NextCloud client.
6. Restart from 1.

I have tried this with another large folder as a new Folder Sync Connection, same setup before. New folder on the client computer, syncing down the remote folder. "use virtual files instead of downloading content immediately" checked on the NC server side.

In this case it would do check the folders on the server side, then just start downloading the whole folder again. No amount of pausing/restarting the client gets it to just create the virtual files instead of syncing everything.
